src/python_testing/TC_OCC_2_2.py

```diff
@@ -0,0 +1,146 @@
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_OCC_2_2(MatterBaseTest):
+    async def read_occ_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.OccupancySensing
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_OCC_2_2(self) -> str:
+        return "[TC-OCC-2.2] OccupancySensorTypeBitmap and OccupancySensorType interdependency with server as DUT"
+
+    def steps_TC_OCC_2_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read OccupancySensorType attribute selection based on FeatureMap Bitmap."),
+            TestStep(3, "Read OccupancySensorTypeBitmap attribute selection based on FeatureMap Bitmap.")
+        ]
+        return steps
+
+    def pics_TC_OCC_2_2(self) -> list[str]:
+        pics = [
+            "OCC.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_OCC_2_2(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        attributes = Clusters.OccupancySensing.Attributes
+        feature_map = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+        is_pir_feature_supported = feature_map & Clusters.OccupancySensing.Bitmaps.Feature.kPassiveInfrared
+        is_us_feature_supported = feature_map & Clusters.OccupancySensing.Bitmaps.Feature.kUltrasonic
+        is_phy_feature_supported = feature_map & Clusters.OccupancySensing.Bitmaps.Feature.kPhysicalContact
+
+        self.step(1)
+        attribute_list = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+
+        self.step(2)
+        # OccupancySensorType will be determined by FeatureMap matching table at 2.7.6.2.
+        if attributes.OccupancySensorType.attribute_id in attribute_list:
+            occupancy_sensor_type_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorType)
+
+            # No Bitmap support from PIR, US, Physical Contact, then OccupancySensorType should be PIR
+            if (is_pir_feature_supported != 1) & (is_us_feature_supported != 1) & (is_phy_feature_supported != 1):

```

Comment:
I don't think this works: you are doing a `&` on a bitmap, so if you do `0b111 & 0b100` for example, the result is 4 (0b100) and not 1.

Why are you not inverting the logic. Like for any specific sensor type you know what should be selected. Like:

```py
if occupancy_sensor_type_dut == Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPir:
   asserts.assert_true(is_pir_feature_supported, "PIR sensor should have the Pir feature")
   asserts.assert_false(is_us_feature_supported, "If both PIR and Ultrasonit is supported, sensor should be PirUltrasonic")
elif occupancy_sensor_type_dut == Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kUltrasonic:
  asserts.assert_true(is_us_feature_supported, "Ultrasonic sensor should have the Ultrasnoic feature")
   asserts.assert_false(is_pir_feature_supported, "If both PIR and Ultrasonit is supported, sensor should be PirUltrasonic")
elif !...
```

or alternatively

```py
assert.assert_equal(
   occupancy_sensor_type_dut == Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kPir,
   is_pir_feature_supported && not is_us_feature_supported
)

assert.assert_equal(
   occupancy_sensor_type_dut == Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kUltrasonic,
   not is_pir_feature_supported && is_us_feature_supported
)

# ...
```